### ISSUE TITLE
[Wave 06] Repository structure + architecture enforcement + regression closure

### ISSUE THREAD

**Issue by bonggyulim** [solar-ai-dev/mcp-work-agent#103]
## Fixed branch / dependency
Work only on `refactor/canonical-architecture-migration`.

Start only from the pushed #102 SHA. Do not run final Gate #97 until this issue is complete, tested, committed, and pushed.

```text
#81 → #98 → #99 → #100 → #101 → #102 → #103 → #97
```

## Mission
Produce the final **repository-structure, single-authority, caller-closure, test-ownership, migration-integrity and regression proof** for Waves #98~#102.

This wave is not a cleanup-by-deletion pass. It must prove both:

1. canonical structure/single authority is actually live; and
2. canonical-compatible implementation/tests/artifacts were preserved during migration unless a responsibility-level reason justified replacement.

## Preservation-first closure rule
For any correction still needed here, apply:

```text
KEEP
→ MOVE / RENAME / MOVE_RENAME
→ SPLIT / MERGE
→ targeted correction
→ REWRITE only when preservation is unsafe
→ DELETE only after caller cut-over
→ CREATE only when no reusable implementation exists
```

`Current code is not authority` does not mean `current code is disposable`.

## Authority
Read actual final files in order:

`00 Project Source Guide → applicable 01~15 concern owner → 04-A/04 where applicable → 16 Source + 16/00..13 → 12-A → 12 → latest Ledger → completed #98~#102 → current code/tests`.

Use Source manifests as the required set. Never derive/widen a canonical manifest from discovered implementation files.

## Verified pre-wave structural baseline — HEAD before #81
Baseline investigated: `2f344e4b93ace4d36a30fd0b6a8d6b12f1036c9c`. Re-verify final closure at #102 HEAD; do not assume every baseline defect still exists.

### A. Confirmed structural debt that Waves must close by preservation-first migration
Baseline contains:

- `adapters/persistence/repositories.py` as a broad ~84 KiB persistence authority alongside canonical owner-specific SQLite repositories;
- `adapters/persistence/unit_of_work.py → SQLiteUnitOfWork` at a noncanonical path/symbol;
- non-empty concrete/root barrel surfaces such as `ports/__init__.py` and `adapters/persistence/__init__.py`;
- many legacy top-level Port contracts (`ports/*.py`, `ports/connectors/**`, `ports/google_oauth.py`, `ports/google_workspace.py`) alongside the newer boundary packages;
- broad legacy runtime `application/coordinator.py → LocalRunCoordinator`;
- broad Connector/runtime files and `google/mcp/workspace_tools.py`/`verified_server.py` alongside the synchronized Connector grammar;
- Frontend feature owners `gmail/tasks/calendar/onboarding/workspace` and broad `frontend/src/api/` rather than the exact current feature-owner grammar;
- top-level `experiments/` containing `candidates/` and `datasets/` while synchronized Source requires the single top-level `evaluation/` root;
- P0 OAuth `GOOGLE_OAUTH_CLIENT_SECRET` / `google_oauth_client_secret` authority in `adapters/connectors/google/mcp/oauth_settings.py`.

These are concrete baseline targets for negative proof, **not permission to delete the directories wholesale**. Verify what #98~#102 moved/merged and whether callers/tests followed.

### B. Confirmed baseline structures that are already correct and must not be rebuilt
At the investigated HEAD:

- `application/workflows/**` does **not** exist — preserve that zero; do not recreate it;
- `NodeRegistry` and `ResumeTargetRegistry` exist at canonical paths and already contain the current 35 runtime-node/profile binding plus exact safe Main resume stages;
- `ResumeTargetRegistry` excludes `ACTION_EXECUTION` and includes `READ_EXECUTION`;
- `CheckpointEffectiveBindingResolver` already delegates target legality to the canonical registry;
- StartRun wire already uses `selected_resource_handles` only;
- production `api/composition.py` already constructs/injects one BackgroundRunExecutor/Schedule/Redrive/RequireRecovery/WorkflowHandoffReconciliationLoop skeleton;
- `WorkflowHandoffReconciliationLoop` is already a thin driver.

The final structure must preserve these correct responsibilities while fixing the remaining gaps. A Wave must not receive credit for deleting/rebuilding an already-canonical authority without necessity.

### 1. Application + Agent operations
Consume only `16/01 Spec → Code Deterministic Mapping` as repository required-operation authority.

Verify:
- Application exact canonical owner/operation/path/symbol/test mappings;
- Agent owner-local operation set = exactly **43**;
- the 43 include `tool_routing.resolve_policy_preconditions` and `planning.resolve_default_container`;
- the 43 Agent repository operations are **not** confused with the separate 35 LangGraph runtime-node adapter manifest;
- required operation = live canonical authority = canonical test owner;
- missing, duplicate or extra live semantic authority = failure.

### 2. Non-persistence Port↔Adapter
Consume only the single **24-row** table in 16/07.

Verify:
- each exact Port path/symbol and P0 Adapter path/symbol;
- `OperationalCommandReplayPort → FilesystemOperationalCommandReplayAdapter` exactly once;
- canonical diagnostic Port is `DiagnosticsPort`; `DiagnosticBundlePort` live authority = 0;
- no parallel old Port-only manifest or public concrete barrel keeps a second authority.

### 3. Runtime identity namespaces
Verify:
- 06-owned runtime stage/node IDs remain unchanged;
- 16-owned Python repository operation symbols remain separate names where mapped;
- no validator forces string equality between runtime ID and Python symbol;
- `ACTION_EXECUTION` remains non-resumable;
- only exact registered Main stages are accepted.

## Caller / import / export / compatibility closure
For every affected capability, produce negative proof for:

- canonical owner live;
- intended API/LangGraph/composition/launcher/other production callers accounted for;
- old production caller count = 0;
- old production import count = 0;
- old concrete export/barrel count = 0;
- duplicate live semantic writer = 0;
- permanent wrapper/compat chain = 0;
- Application concrete Adapter/SQLite/Provider dependency = 0;
- Adapter reverse-import of Application concrete semantic authority = 0;
- Domain→Application/API/Adapter forbidden imports = 0;
- LangGraph node direct SQLite/provider/Domain-transition implementation = 0;
- concrete production construction outside `api/composition.py` = 0 except exact current Exception Registry entries.

A stable public contract/Port re-export is not automatically a compatibility violation. A concrete Handler/Adapter/service/runtime/facade re-export is.

## Top-level root closure
Verify final roots against 16/02:

- current evaluation implementation/data/artifacts are under top-level `evaluation/`;
- top-level `experiments/` = 0 after its candidates/datasets/artifacts are preservation-first MOVED/MERGED into `evaluation/**` according to 13/16;
- do not throw away experiment datasets/candidates to satisfy the root-name check;
- product runtime imports `evaluation/**` = 0;
- product runtime imports installer/release sources = 0.

## Canonical test ownership and strength
For migrated responsibilities prove:

- unit tests mirror canonical production owner paths;
- frontend tests use exact `frontend/tests/app/**` / `frontend/tests/features/<owner>/**` manifest paths;
- legacy-path tests are not the primary behavior owner;
- moved/split/merged code retains equivalent or stronger behavioral coverage;
- legal/illegal transition matrices, CAS/concurrency, crash cuts, security negatives and replay conflicts exercise the real boundary;
- source-string scans alone are not behavioral proof;
- mocks are not weaker than actual Protocol/UoW/CAS contracts;
- no assertion deletion, expected-result weakening, raw SQL fixture bypass or allowlist widening hides regression.

## Migration integrity
At final #103 HEAD prove:

- applied migrations were never renamed/rewritten/reordered/squashed;
- exact populated `0009→0010` and `0010→0011` upgrades pass;
- each new migration added by #99 passes from its immediately previous populated version;
- prior checksum/name/order unchanged;
- post-upgrade indexes/constraints/data/receipts/foreign-key integrity pass;
- migration-count-only assertions are insufficient.

## Preservation evidence — required
For every responsibility touched by #98~#102, closure matrix must include:

- original path/symbol;
- canonical path/symbol;
- production callers before/after;
- disposition;
- reused implementation/test/fixture reference;
- targeted semantic correction if any;
- `REWRITE/DELETE/CREATE` justification;
- caller-cutover proof before deletion;
- behavior-regression test result.

BLOCK this Wave if:
- path/name mismatch was implemented as equivalent delete+recreate without need;
- a broad file was discarded despite reusable canonical logic;
- a compatibility facade/old authority was deleted before its callers or useful implementation were migrated;
- a genuinely unmigrated capability was deleted only to satisfy a negative scan;
- REWRITE/DELETE/CREATE lacks responsibility-level evidence.

## Cumulative regression execution
At the single final #103 HEAD run all available affected:

- unit;
- contract;
- integration;
- component/frontend;
- workflow/runtime/crash;
- security/OAuth/Connector;
- migration;
- architecture/enforcement;
- Ruff/mypy/typecheck/build as applicable;
- `git diff --check`.

For full-suite failures, run the identical command at the exact cumulative comparison base and classify regression vs pre-existing baseline. Report NOT_EXECUTED when not run.

## Required closure matrix columns

```text
capability / Ledger row
canonical authority
original implementation source
intended callers
observed canonical callers
old caller count
old import/export count
duplicate writer count
compat/wrapper count
canonical test owner
disposition
preserved/moved/split/merged references
REWRITE/DELETE/CREATE justification
targeted test command/result
unresolved debt classification
```

This matrix is verification output, not a second canonical manifest.

## Done condition
At one pushed revision:

- exact 43 Agent-operation and 24 Port↔Adapter source manifests are singular and enforced;
- exact 35 runtime-node identity mapping remains separate and correct;
- caller/import/export/compat/duplicate negative proof passes;
- top-level `experiments/` = 0 with artifacts preserved under canonical evaluation ownership;
- architecture exception drift = 0;
- canonical test-owner gaps = 0;
- applied migration mutation = 0 and populated upgrades PASS;
- client-secret authority = 0;
- broad persistence/coordinator/connector/frontend legacy authorities = 0 for migrated responsibilities after caller cut-over;
- regression-hiding changes = 0;
- unjustified implementation/test/fixture loss = 0;
- unjustified equivalent `DELETE + CREATE` churn = 0;
- all required available targeted/architecture suites PASS.

## Completion report
Return the closure matrix, files moved/split/merged/deleted, preservation evidence, exact commands/results, baseline comparisons, unresolved debt, and pushed SHA.

**Comment by bonggyulim** [comment]
Superseded by #104. Repository-wide closure/enforcement must not run ahead of unresolved Domain authority. Recreate/re-scope the final structural closure issue only after the Domain layer is fully closed at one pushed revision.
